myaws/ecs_node_renew.go

```diff
@@ -118,6 +119,42 @@ func (client *Client) ECSNodeRenew(options ECSNodeRenewOptions) error {
 		return err
 	}
 
+	// Get a list of instance IDs before auto scaling
+	var oldInstanceIds []*string
+	for _, oldNode := range oldNodes {
+		oldInstanceIds = append(oldInstanceIds, oldNode.Ec2InstanceId)
+	}
+
+	// Get a list of instances after auto scaling
+	allNodes, err := client.findECSNodes(options.Cluster)
+	if err != nil {
+		return err
+	}
+
+	// Get a list of instance IDs after auto scaling
+	var allInstanceIds []*string
+	for _, allNode := range allNodes {
+		allInstanceIds = append(allInstanceIds, allNode.Ec2InstanceId)
+	}
```

Comment:
Since the main logic here becomes long, can we move the getting `oldInstanceIds` and `allInstanceIds` logic into the `selectInstanceToProtectFromScaleIn` func?

Reply:
fixed, thank you.